package/MDAnalysis/analysis/msd.py

```diff
@@ -288,9 +290,12 @@ def __init__(self, u, select, msd_type='xyz', fft=True, **kwargs):
             Defaults to ``True``.    
 
         """
+        if isinstance(u, groups.UpdatingAtomGroup):
+            raise TypeError("UpdatingAtomGroups are not valid for MSD "
+                            "computation")
         self.u = u
```

Comment:
Is this actually stored anywhere but line 302? I would suggest cleaning up the attributes list and just doing self._atoms = u.select_atoms(select). 

Although with hindsight, I'd suggest naming that attribute `self._atomgroup` (like the density code) or even going for a "non-private" version like `self.ag` like we do in places like Dihedrals and Hole2.

Beyond the scope of this conversion, we really should "unify" the way we store different things. I think there's a reasonable case for expecting that things are consistently stored/named across AnalysisBase classes.